src/Files.App/Helpers/ZipHelpers.cs

```diff
@@ -6,125 +6,182 @@
-                {
-                    if (!NativeFileOperationsHelper.CreateDirectoryFromApp(dir, IntPtr.Zero))
-                    {
-                        var dirName = destinationFolder.Path;
-                        foreach (var component in dir.Split(Path.DirectorySeparatorChar, StringSplitOptions.RemoveEmptyEntries))
-                        {
-                            dirName = Path.Combine(dirName, component);
-                            NativeFileOperationsHelper.CreateDirectoryFromApp(dirName, IntPtr.Zero);
-                        }
-                    }
-
-                    if (cancellationToken.IsCancellationRequested) // Check if canceled
-                    {
-                        return;
-                    }
-                }
-
-                if (cancellationToken.IsCancellationRequested) // Check if canceled
-                {
-                    return;
-                }
-
-                // Fill files
-
-                byte[] buffer = new byte[4096];
-                int entriesAmount = fileEntries.Count;
-                int entriesFinished = 0;
-
-                foreach (var entry in fileEntries)
-                {
-                    if (cancellationToken.IsCancellationRequested) // Check if canceled
-                    {
-                        return;
-                    }
-                    if (entry.Encrypted)
-                    {
-                        App.Logger.Info($"Skipped encrypted zip entry: {entry.FileName}");
-                        continue; // TODO: support password protected archives
-                    }
-
-                    string filePath = Path.Combine(destinationFolder.Path, entry.FileName);
-
-                    var hFile = NativeFileOperationsHelper.CreateFileForWrite(filePath);
-                    if (hFile.IsInvalid)
-                    {
-                        return; // TODO: handle error
-                    }
-
-                    // We don't close hFile because FileStream.Dispose() already does that
-                    using (FileStream destinationStream = new FileStream(hFile, FileAccess.Write))
-                    {
-                        try
-                        {
-                            await zipFile.ExtractFileAsync(entry.Index, destinationStream);
-                        }
-                        catch (Exception ex)
-                        {
-                            App.Logger.Warn(ex, $"Error extracting file: {filePath}");
-                            return; // TODO: handle error
-                        }
-                    }
-
-                    entriesFinished++;
-                    float percentage = (float)((float)entriesFinished / (float)entriesAmount) * 100.0f;
-                    progressDelegate?.Report(percentage);
-                }
-            }
-        }
-    }
+	public static class ZipHelpers
+	{
+		public static async Task<bool> CompressSingleToArchive(string sourceFolder, string archive)
```

Comment:
Is it possible to have a single task that handles this logic? Ideally we can pass in sourceFolder or sourceFolders to the same task.